backend/opendrive/opendrive.go

```diff
@@ -1027,6 +1033,32 @@ func (o *Object) readMetaData(ctx context.Context) (err error) {
 		return err
 	}
 	var resp *http.Response
+
+	// If we know the object id perform a direct lookup
+	// because the /folder/itembyname.json endpoint is unreliable:
+	// newly created objects take an arbitrary amount of time to show up
+	if "" != o.id {
+		fileInfo := File{}
+		err = o.fs.pacer.Call(func() (bool, error) {
+			opts := rest.Opts{
+				Method: "GET",
+				Path: fmt.Sprintf("/file/info.json/%s?session_id=%s",
+					o.id, o.fs.session.SessionID),
+			}
+			resp, err = o.fs.srv.CallJSON(ctx, &opts, nil, &fileInfo)
+			return o.fs.shouldRetry(ctx, resp, err)
+		})
+		if err != nil {
+			return fmt.Errorf("failed to get fileinfo: %w", err)
+		}
+
+		o.id = fileInfo.FileID
```

Comment:
This feels like this should be an `if / else` with the setting of the `o.id` etc variables done after the `if / else` otherwise there is a bit of duplicated code here.

Reply:
done!